targets/TARGET_STM/trng_api.c

```diff
@@ -63,23 +66,31 @@ void trng_free(trng_t *obj)
     HAL_RNG_DeInit(&obj->handle);
     /* RNG Peripheral clock disable - assume we're the only users of RNG  */
     __HAL_RCC_RNG_CLK_DISABLE();
+
+    users = 0;
 }
 
 int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
 {
-    int ret;
+    int ret = 0;
+    uint8_t random[4];
+    *output_length = 0;
 
     /* Get Random byte */
-    for( uint32_t i = 0; i < length; i++ ){
-        trng_get_byte(obj, output + i );
+    while ((*output_length < length) && (ret ==0)) {
+        if ( HAL_RNG_GenerateRandomNumber(&obj->handle, (uint32_t *)random ) != HAL_OK) {
+                ret = -1;
+        } else {
+            for (uint8_t i =0; i < 4; i++) {
+                *output++ = random[i];
```

Comment:
what about corner cases where output length will not be a multiple of 4? In this case you will overflow the buffer

Reply:
I thought I'd covered it, but clearly missed here - will fix - thx Ron  !